### PR TITLE
Change Salary's regex to allow only maximum of 9 digits and disallow

### PR DESCRIPTION
**Pull request by e0543517**
Change Salary's regex to allow only maximum of 9 digits and disallow salary starting with 0

### CODE REVIEW

File: src/main/java/seedu/address/model/job/Salary.java [hunk 1]
```diff
@@ -7,12 +7,14 @@
     public static final String MESSAGE_CONSTRAINTS = "Salary must be a string with the format "
             + "[Non-negative integers][-][Non-negative integers]\n"
             + "The right bound salary should be equal to or more than the left bound salary\n"
+            + "The maximum number of digits for the salary is 9 digits \n"
+            + "The salary cannot start with a 0 eg 01234 - 0032134\n"
             + "Salary is salary paid per month and in SGD\n"
             + "eg 4000-5000";
 
-    public static final String LEFT_REGEX = "^[ ]*[0-9]+[ ]*";
+    public static final String LEFT_REGEX = "^[ ]*[1-9][0-9]{0,8}+[ ]*";
```
Comment (Justinnje) [inline src/main/java/seedu/address/model/job/Salary.java:15]: Can write some tests to test that if there are 10 numbers, it will throw an error.

**Review by Justinnje — COMMENTED** [review]
LGTM just write one or two test and should be good

File: src/main/java/seedu/address/model/job/Salary.java (hunk 1 above)
Comment (seanleong339) [reply src/main/java/seedu/address/model/job/Salary.java:15]: Yea just some simple changes in SalaryTest will be sufficient

**Review by seanleong339 — COMMENTED** [review]
LGTM, but can do some simple test in SalaryTest

File: src/main/java/seedu/address/model/job/Salary.java (hunk 1 above)
Comment (e0543517) [reply src/main/java/seedu/address/model/job/Salary.java:15]: Updated test cases. Please review again.

**Review by Justinnje — APPROVED** [review]
LGTM

**Pull request merged**